packages/components/src/Link/Link.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { forwardRef } from 'react'
+import { Link as RACLink, type LinkProps as RACLinkProps } from 'react-aria-components'
+import { type TextProps } from '~components/Text'
+import { mergeClassNames } from '~components/utils/mergeClassNames'
+import { LinkContent } from './subcomponents/LinkContent'
+import styles from './Link.module.css'
+
+export type LinkProps = (UnderlinedLink | NonUnderlinedLink) & (InlineLink | NonInlineLink)
+
+type BaseLinkProps = {
+  variant?: 'primary' | 'secondary' | 'white'
+  iconPosition?: 'start' | 'end'
+} & Omit<RACLinkProps, 'children'> & {
+    /** Used as the label for the Link. */
+    children: RACLinkProps['children']
+  }
+
+export type UnderlinedLink = BaseLinkProps & {
```

Comment:
One thing I noticed is that we're create a new intersection with `BaseLinkProps` in each of these types, then in the `LinkProps` we're combining all of these together again.

![Screenshot 2025-02-17 at 2 03 14 pm](https://github.com/user-attachments/assets/3e455f45-a26c-4864-acfe-9d7fdd45a660)

 I'm not sure the potential impact on the final type but we might be able to reduce the times that `BaseLinkProps` is added into our types and intersect it in the `LinkProps` instead